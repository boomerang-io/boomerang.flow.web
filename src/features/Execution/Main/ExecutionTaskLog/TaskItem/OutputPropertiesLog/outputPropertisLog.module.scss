@import "~Styles/mixins";

.container {
  max-width: 50rem;

  :global .bx--modal-content {
    padding-right: 1rem;

    .bx--tab-content {
      padding: 0;
    }
  }
}

.trigger {
  color: var(--cds-interactive-01);
  cursor: pointer;
  font-size: 0.875rem;
  line-height: 1.125rem;

  &:focus {
    @include focus-outline();
  }
}

.propertiesJson {
  background-color: var(--cds-ui-01);
  height: 25rem;
  overflow-y: auto;
  margin-top: 1rem;

  :global .object-key {
    span {
      color: var(--cds-interactive-01);
    }
  }
}
