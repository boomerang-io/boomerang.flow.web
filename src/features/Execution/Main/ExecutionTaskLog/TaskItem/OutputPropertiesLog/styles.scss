@import "~@boomerang/boomerang-styles/lib/colors";
@import "~Styles/mixins";
@import "~Styles/variables";

.c-output-properties {
  max-width: 50rem;
}

.s-output-properties-trigger {
  color: var(--cds-interactive-01);
  cursor: pointer;
  font-size: 0.875rem;
  line-height: 1.125rem;

  &:focus {
    @include focus-outline();
  }
}

.s-output-properties-json {
  background-color: $white;
  height: 25rem;
  overflow-y: auto;
  margin-top: 1rem;

  .object-key {
    span {
      color: var(--cds-interactive-01);
    }
  }
}
