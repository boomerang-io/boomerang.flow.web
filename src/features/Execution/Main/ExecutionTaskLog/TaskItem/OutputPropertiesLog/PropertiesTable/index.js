import React from "react";
import PropTypes from "prop-types";
import { DataTable } from "carbon-components-react";
import { NoDisplay } from "@boomerang/carbon-addons-boomerang-react";
import styles from "./propertiesTable.module.scss";

function PropertiesTable({ data: properties }) {
  const headers = [{ header: "Property", key: "key" }, { header: "Value", key: "value" }];

  const renderCell = (cellIndex, value) => {
    const column = headers[cellIndex];

    switch (column.header) {
      case "Value":
        return <code className={styles.code}>{value || "---"}</code>;
      default:
        return <p className={styles.tableTextarea}>{value || "---"}</p>;
    }
  };

  const { TableContainer, Table, TableHead, TableRow, TableBody, TableCell, TableHeader } = DataTable;

  return (
    <div className={styles.tableContainer}>
      {properties && properties.length > 0 ? (
        <DataTable
          rows={properties}
          headers={headers}
          render={({ rows, headers, getHeaderProps }) => (
            <TableContainer>
              <Table>
                <TableHead>
                  <TableRow className={styles.tableHeadRow}>
                    {headers.map(header => (
                      <TableHeader
                        id={header.key}
                        {...getHeaderProps({
                          header,
                          className: `${styles.tableHeadHeader} ${styles[header.key]}`
                        })}
                      >
                        {header.header}
                      </TableHeader>
                    ))}
                  </TableRow>
                </TableHead>
                <TableBody className={styles.tableBody}>
                  {rows.map(row => (
                    <TableRow key={row.id} className={styles.tableRow}>
                      {row.cells.map((cell, cellIndex) => (
                        <TableCell key={cell.id} style={{ padding: "0" }}>
                          <div className={styles.tableCell}>{renderCell(cellIndex, cell.value)}</div>
                        </TableCell>
                      ))}
                    </TableRow>
                  ))}
                </TableBody>
              </Table>
            </TableContainer>
          )}
        />
      ) : (
        <NoDisplay text="No properties to display" />
      )}
    </div>
  );
}

PropertiesTable.propTypes = {
  data: PropTypes.array.isRequired
};

export default PropertiesTable;
