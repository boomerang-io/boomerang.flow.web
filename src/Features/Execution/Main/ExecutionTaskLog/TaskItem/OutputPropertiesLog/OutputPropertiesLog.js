import React from "react";
import PropTypes from "prop-types";
import ReactJson from "react-json-view";
import { Button, ComposedModal, ModalForm } from "@boomerang-io/carbon-addons-boomerang-react";
import { ModalBody, Tabs, Tab } from "@boomerang-io/carbon-addons-boomerang-react";
import PropertiesTable from "./PropertiesTable";
import styles from "./outputPropertisLog.module.scss";

function OutputPropertiesLog({ flowTaskName, flowTaskOutputs, isOutput }) {
  let arrayProps = [];
  Object.keys(flowTaskOutputs).forEach(
    (val, index) =>
      (arrayProps = arrayProps.concat({
        id: `${val}-${index}`,
        key: val,
        value: JSON.stringify(flowTaskOutputs[val], null, 2),
      }))
  );

  return (
    <ComposedModal
      composedModalProps={{ containerClassName: styles.container, shouldCloseOnOverlayClick: true }}
      confirmModalProps={{
        title: "Are you sure?",
        children: "Your changes will not be saved",
      }}
      modalHeaderProps={{
        title: "Output Properties",
        label: `${flowTaskName}`,
      }}
      modalTrigger={({ openModal }) => (
        <Button kind="ghost" size="small" onClick={openModal}>
          View Properties
        </Button>
      )}
    >
      {() => (
        <ModalForm>
          <ModalBody>
            <Tabs>
              <Tab label="Table">
                <PropertiesTable hasJsonValues={!isOutput} data={isOutput ? flowTaskOutputs : arrayProps} />
              </Tab>
              <Tab label="JSON">
                <div className={styles.propertiesJson}>
                  <ReactJson
                    name={false}
                    src={flowTaskOutputs}
                    displayDataTypes={false}
                    enableDelete={false}
                    displayObjectSize={false}
                    enableEdit={false}
                    enableAdd={false}
                  />
                </div>
              </Tab>
            </Tabs>
          </ModalBody>
        </ModalForm>
      )}
    </ComposedModal>
  );
}

OutputPropertiesLog.propTypes = {
  flowTaskName: PropTypes.string.isRequired,
  flowTaskOutputs: PropTypes.object.isRequired,
};

export default OutputPropertiesLog;
