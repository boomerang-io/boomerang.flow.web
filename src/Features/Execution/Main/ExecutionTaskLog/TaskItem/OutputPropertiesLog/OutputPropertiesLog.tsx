import React from "react";
import ReactJson from "react-json-view";
import { ComposedModal, ModalForm } from "@boomerang-io/carbon-addons-boomerang-react";
import { ModalBody, Tabs, Tab } from "@boomerang-io/carbon-addons-boomerang-react";
import PropertiesTable from "./PropertiesTable";
import { ModalTriggerProps } from "Types";

import styles from "./outputPropertisLog.module.scss";

interface OutputPropertiesLogProps {
  flowTaskName: string;
  flowTaskOutputs: object[] | {};
}

const OutputPropertiesLog: React.FC<OutputPropertiesLogProps> = ({ flowTaskName, flowTaskOutputs }) => {
  let arrayProps: object[] = [];
  Object.keys(flowTaskOutputs).forEach(
    (val, index) =>
      (arrayProps = arrayProps.concat({
        id: `${val}-${index}`,
        key: val,
        //@ts-ignore
        value: JSON.stringify(flowTaskOutputs[val], null, 2),
      }))
  );

  return (
    <ComposedModal
      composedModalProps={{ containerClassName: styles.container, shouldCloseOnOverlayClick: true }}
      confirmModalProps={{
        title: "Are you sure?",
        children: "Your changes will not be saved",
      }}
      modalHeaderProps={{
        title: "Output Properties",
        label: `${flowTaskName}`,
      }}
      modalTrigger={({ openModal }: ModalTriggerProps) => (
        <button className={styles.trigger} onClick={openModal}>
          View Properties
        </button>
      )}
    >
      {() => (
        <ModalForm>
          <ModalBody>
            <Tabs>
              <Tab label="Table">
                <PropertiesTable properties={arrayProps} />
              </Tab>
              <Tab label="JSON">
                <div className={styles.propertiesJson}>
                  <ReactJson
                    name={false}
                    src={flowTaskOutputs}
                    displayDataTypes={false}
                    // enableDelete={false}
                    displayObjectSize={false}
                    // enableEdit={false}
                    // enableAdd={false}
                  />
                </div>
              </Tab>
            </Tabs>
          </ModalBody>
        </ModalForm>
      )}
    </ComposedModal>
  );
};

export default OutputPropertiesLog;
