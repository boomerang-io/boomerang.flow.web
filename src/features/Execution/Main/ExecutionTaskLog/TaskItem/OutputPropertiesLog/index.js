import React from "react";
import PropTypes from "prop-types";
import ReactJson from "react-json-view";
import { ModalFlow, ModalFlowForm } from "@boomerang/carbon-addons-boomerang-react";
import { ModalBody, Tabs, Tab } from "carbon-components-react";
import PropertiesTable from "./PropertiesTable";
import "./styles.scss";

function OutputPropertiesLog({ flowTaskName, flowTaskOutputs }) {
  let arrayProps = [];
  Object.keys(flowTaskOutputs).forEach(
    (val, index) =>
      (arrayProps = arrayProps.concat({
        id: `${val}-${index}`,
        key: val,
        value: JSON.stringify(flowTaskOutputs[val], null, 2)
      }))
  );

  return (
    <ModalFlow
      composedModalProps={{ containerClassName: "c-output-properties" }}
      confirmModalProps={{
        title: "Are you sure?",
        children: "Your changes will not be saved"
      }}
      modalHeaderProps={{
        title: "Output Properties",
        label: `${flowTaskName}`
      }}
      modalTrigger={({ openModal }) => (
        <button className="s-output-properties-trigger" onClick={openModal}>
          View Properties
        </button>
      )}
    >
      <ModalFlowForm>
        <ModalBody>
          <Tabs>
            <Tab label="Table">
              <PropertiesTable data={arrayProps} />
            </Tab>
            <Tab label="JSON">
              <div className="s-output-properties-json">
                <ReactJson
                  name={false}
                  src={flowTaskOutputs}
                  displayDataTypes={false}
                  enableDelete={false}
                  displayObjectSize={false}
                  enableEdit={false}
                  enableAdd={false}
                />
              </div>
            </Tab>
          </Tabs>
        </ModalBody>
      </ModalFlowForm>
    </ModalFlow>
  );
}

OutputPropertiesLog.propTypes = {
  flowTaskName: PropTypes.string.isRequired,
  flowTaskOutputs: PropTypes.object.isRequired
};

export default OutputPropertiesLog;
