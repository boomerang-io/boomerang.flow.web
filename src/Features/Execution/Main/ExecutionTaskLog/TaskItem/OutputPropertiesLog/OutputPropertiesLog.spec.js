import React from "react";
import OutputPropertiesLog from "./index";
import { queryCaches } from "react-query";

afterEach(() => {
  queryCaches.forEach((queryCache) => queryCache.clear());
});

const props = {
  flowTaskName: "Send Slack Message",
  flowTaskOutputs: {
    args: {
      test: "true",
    },
  },
};

describe("OutputPropertiesLog --- Snapshot", () => {
  it("Capturing Snapshot of OutputPropertiesLog", () => {
    const { baseElement } = rtlRender(<OutputPropertiesLog {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
