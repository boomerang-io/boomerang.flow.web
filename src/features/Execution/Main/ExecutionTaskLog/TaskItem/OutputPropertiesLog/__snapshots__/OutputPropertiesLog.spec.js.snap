// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutputPropertiesLog --- Snapshot Capturing Snapshot of OutputPropertiesLog 1`] = `
<body
  id="app"
>
  <div>
    <button
      class="trigger"
    >
      View Properties
    </button>
  </div>
  <div
    class="bx--bmrg-modal-portal"
  />
</body>
`;
